SmartDeviceLink/SDLChoiceSet.m

```diff
@@ -62,23 +62,29 @@ - (instancetype)initWithTitle:(NSString *)title delegate:(id<SDLChoiceSetDelegat
     }
 
     NSMutableSet<NSString *> *choiceTextSet = [NSMutableSet setWithCapacity:choices.count];
+    NSMutableSet<NSString *> *allMenuVoiceCommands = [NSMutableSet set];
+    NSUInteger allVoiceCommandsCount = 0;
+    NSUInteger choiceSetWithVoiceCommandCount = 0;
     for (SDLChoiceCell *cell in choices) {
         [choiceTextSet addObject:cell.text];
+        if (cell.voiceCommands == nil) { continue; }
+        [allMenuVoiceCommands addObjectsFromArray:cell.voiceCommands];
+        choiceSetWithVoiceCommandCount += 1;
+        allVoiceCommandsCount += cell.voiceCommands.count;
     }
     if (choiceTextSet.count < choices.count) {
-        SDLLogW(@"Attempted to create a choice set with duplicate cell text. Cell text must be unique.");
+        SDLLogE(@"Attempted to create a choice set with duplicate cell text. Cell text must be unique. The choice set will not be set.");
         return nil;
     }
 
-    NSMutableArray<NSArray<NSString *> *> *nonNilVoiceCommands = [NSMutableArray array];
-    for (SDLChoiceCell *cell in choices) {
-        if (cell.voiceCommands != nil) {
-            [nonNilVoiceCommands addObject:cell.voiceCommands];
-        }
+    // All or none of the choices must have VR commands
+    if ((choiceSetWithVoiceCommandCount > 0 && choiceSetWithVoiceCommandCount < choices.count)) {
+        SDLLogE(@"If using voice recognition commands, all of the choice set cells must have unique VR commands. There are %lu unique VR commands and %lu choice set items. The choice set will not be set.", (unsigned long)choiceSetWithVoiceCommandCount, (unsigned long)choices.count);
```

Comment:
"...There are %lu cells with unique voice commands and %lu total cells"

Reply:
Fixed